packages/sources/coingecko/src/endpoint/crypto.ts

```diff
@@ -64,21 +65,20 @@ const handleBatchedRequest = (
   jobRunID: string,
   request: AdapterRequest,
   response: AxiosResponse<ResponseSchema>,
-  validator: Validator,
   endpoint: string,
-  idToSymbol: Record<string, string>,
+  idToSymbol: OverrideToOriginalSymbol,
 ) => {
   const payload: [AdapterRequest, number][] = []
-  for (const base in response.data) {
+  for (const base of response.data) {
     const quoteArray = Array.isArray(request.data.quote) ? request.data.quote : [request.data.quote]
     for (const quote of quoteArray) {
-      const symbol = idToSymbol?.[base]
-      if (symbol) {
+      const originalSymbol = idToSymbol[base]
+      if (originalSymbol) {
         const individualRequest = {
           ...request,
           data: {
             ...request.data,
-            base: validator.overrideReverseLookup(AdapterName, 'overrides', symbol).toUpperCase(),
+            base: idToSymbol[originalSymbol].toUpperCase(),
```

Comment:
This is returning `undefined` and then throwing when attempting to call the `.toUpperCase` method.

Reply:
Fixed this and added test for array of symbols